File: app/src/main/java/com/github/malitsplus/shizurunotes/common/I18N.java
package com.github.malitsplus.shizurunotes.common;

import android.app.Application;

import androidx.annotation.ArrayRes;
import androidx.annotation.StringRes;

import com.github.malitsplus.shizurunotes.R;

import java.util.Arrays;

public class I18N {

    public static Application application;

    private I18N(){
    }

    public static String getString(@StringRes int resId){
        if(application != null)
            return application.getString(resId);
        else
            return "";
    }

    public static String[] getStringArray(@ArrayRes int resId){
        if(application != null)
            return application.getResources().getStringArray(resId);
        else
            return new String[] {""};
    }

    public static String getStringWithSpace(@StringRes int resId){
        if (application != null)
            return application.getString(R.string.space_modifier, application.getString(resId));
        else
            return "";
    }

    public static String getString(@StringRes int resId, Object... formatArgs){
        if(application != null)
            return application.getString(resId, formatArgs);
        else
            return "";
    }
}

File: app/src/main/java/com/github/nyanfantasia/shizurunotes/common/I18N.kt
package com.github.nyanfantasia.shizurunotes.common

import android.app.Application
import androidx.annotation.ArrayRes
import androidx.annotation.StringRes
import com.github.nyanfantasia.shizurunotes.R

class I18N {
    companion object {
        var application: Application? = null
        fun getString(@StringRes resId: Int): String {
            return if (application != null) application!!.getString(
                resId
            ) else ""
        }

        fun getStringArray(@ArrayRes resId: Int): Array<String> {
            return if (application != null) application!!.resources.getStringArray(
                resId
            ) else arrayOf("")
        }

        @JvmStatic
        fun getStringWithSpace(@StringRes resId: Int): String {
            return if (application != null) application!!.getString(
                R.string.space_modifier,
                application!!.getString(resId)
            ) else ""
        }

        @JvmStatic
        fun getString(@StringRes resId: Int, vararg formatArgs: Any?): String {
            return if (application != null) application!!.getString(
                resId,
                *formatArgs
            ) else ""
        }
    }
}